src/fs/mod.rs

```diff
@@ -323,13 +323,20 @@ where
     }
 
     /// Create this controller
-    fn create(&self) {
+    fn create(&self) -> Result<()> {
         self.verify_path()
             .unwrap_or_else(|_| panic!("path should be valid: {:?}", self.path()));
 
         match ::std::fs::create_dir_all(self.get_path()) {
-            Ok(_) => self.post_create(),
-            Err(e) => warn!("error create_dir: {:?} error: {:?}", self.get_path(), e),
+            Ok(_) => {
+                self.post_create();
+                Ok(())
+            }
+            Err(e) => {
+                let err_msg = format!("error create_dir: {:?}, because :{:?}", self.get_path(), e);
+                warn!("{}", err_msg);
+                Err(Error::with_cause(ErrorKind::Common(err_msg), e))
+            }
         }
```

Comment:
Don't need to use `match`, and the error if any is thrown, don't need to log a warning as well.

```rust
std::fs::create_dir_all(self.get_path()).map_err(|err| // blabla)?;
self.post_create();

Ok(())
```